src/main/java/pl/tomaszdziurko/jvm_bloggers/blogs/BloggersDataUpdater.java

```diff
@@ -55,22 +53,28 @@ protected void updateSingleEntry(BloggerEntry bloggerEntry, UpdateSummary update
     private void updateBloggerIfThereAreSomeChanges(BloggerEntry bloggerEntry,
                                                     UpdateSummary updateSummary,
                                                     Blog existingBlogger) {
-        bloggerEntry.setUrl(
-            syndFeedFactory.validUrlFromRss(bloggerEntry.getRss()).orElse(existingBlogger.getUrl())
-        );
-        
-        if (!isEqual(existingBlogger, bloggerEntry)) {
+
+        Optional<String> validBlogUrl = extractValidBlogUrlFromFeed(bloggerEntry.getRss());
+        validBlogUrl.ifPresent(bloggerEntry::setUrl);
+
+        if (somethingChangedInBloggerData(existingBlogger, bloggerEntry)) {
             existingBlogger.setJsonId(bloggerEntry.getJsonId());
             existingBlogger.setAuthor(bloggerEntry.getName());
             existingBlogger.setTwitter(bloggerEntry.getTwitter());
             existingBlogger.setRss(bloggerEntry.getRss());
             existingBlogger.setBlogType(bloggerEntry.getBlogType());
-            existingBlogger.setUrl(bloggerEntry.getUrl());
+            if (bloggerEntry.getUrl() != null) {
+                existingBlogger.setUrl(bloggerEntry.getUrl());
+            }
             blogRepository.save(existingBlogger);
             updateSummary.recordUpdated();
         }
     }
 
+    private Optional<String> extractValidBlogUrlFromFeed(String rss) {
```

Comment:
\[Checkstyle\] WARNING: 'method def modifier' have incorrect indentation level 4, expected level should be 2\.